### PR TITLE
avoid dropping deps file when publishing

### LINKED ISSUE #157: dotnet publish should not output deps file
@anurse Is the deps file preferred over app local? If so then that might post problematic for our current build and things might just be working by fluke. 

Here's what the output folder looks like:

image

### PR DESCRIPTION
**Pull request by anurse**
also, avoid requiring a dependency on Microsoft.NETCore.Runtime

I also added a script that allows one-time invocation of the dotnet toolset in your output dir. Just run `scripts/dev-dotnet.sh [cmd] [args]`/`scripts\dev-dotnet.ps1 [cmd] [args]`

fixes #157 

/cc @davidfowl @Sridhar-MS @brthor @piotrpMSFT

### CODE REVIEW

**Comment by anurse** [comment]
@dotnet-bot test this please

**Comment by Sridhar-MS** [comment]
@dotnet-bot test this please

File: scripts/dev-dotnet.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (Sridhar-MS) [inline scripts/dev-dotnet.sh]: I dont see the dev-dotnet scripts getting used anywhere.

Comment (Sridhar-MS) [inline scripts/dev-dotnet.sh]: Ignore this.. I just saw your PR description.

File: src/Microsoft.DotNet.Tools.Publish/Program.cs [hunk 2]
```diff
@@ -170,8 +175,34 @@ private static int PublishHost(ProjectContext context, string outputPath)
 
             // Copy the host
             File.Copy(hostPath, outputExe, overwrite: true);
+
+            // Publish the runtime
+            var runtimePath = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "..", "runtime", "coreclr"));
+            if(!Directory.Exists(runtimePath))
```
Comment (Sridhar-MS) [inline src/Microsoft.DotNet.Tools.Publish/Program.cs]: This is going to override any direct reference to the Runtime that the project has. This will prevent the project from using specific versions of the runtime.

Comment (anurse) [inline src/Microsoft.DotNet.Tools.Publish/Program.cs]: Correct. I'm not sure the right approach there since I'm really not sure having a dependency on the Runtime is the right way to do this anyway. Apps shouldn't be dependent on specific runtime versions as much as possible, that should be a deployment-time decision.

I considered having publish take a `--runtime-path` switch to allow the caller to set the source path for runtime files.

Comment (Sridhar-MS) [inline src/Microsoft.DotNet.Tools.Publish/Program.cs]: I think the right approach is if the runtime is published already via a direct reference then dont overwrite it.

File: src/Microsoft.DotNet.Tools.Compiler/Program.cs [hunk 3]
```diff
@@ -361,7 +369,7 @@ private static bool CompileProject(ProjectContext context, string configuration,
 
             var success = result.ExitCode == 0;
 
-            if (success && compilationOptions.EmitEntryPoint.GetValueOrDefault())
+            if (success && !noHost && compilationOptions.EmitEntryPoint.GetValueOrDefault())
```
Comment (Sridhar-MS) [inline src/Microsoft.DotNet.Tools.Compiler/Program.cs]: This is going to prevent copying the project-to-project dependencies as well. Is that ok? Does `publish` take care of them too when publishing.

Comment (anurse) [inline src/Microsoft.DotNet.Tools.Compiler/Program.cs]: Yes, publish with take care of project-to-project dependencies.

**Comment by krwq** [comment]
@anurse, could you make sure @piotrpMSFT takes a look at it? I believe he was touching .deps files - let's make sure we are not interfering with each other

**Comment by piotrpMSFT** [comment]
no deps in publish LGTM. These mechanisms are not intended to be used together.

@anurse @Sridhar-MS can you guys pull together a strawman for how were going to deal with runtime references? We need to handle a few scenarions:
-  by default "it just works"
- cross-targeting builds [with|out native dependencies]
- specifying an exact runtime [Fx Devs]

**Comment by anurse** [comment]
I'm going to remove that piece of the publish touch-ups and continue with our existing story (have to have a dependency on it) for now.

File: scripts/dev-dotnet.sh (hunk 1 above)
Comment (anurse) [inline scripts/dev-dotnet.sh]: Yeah, it was just a utility script I wrote while testing this PR :). It could be in it's own PR but I think that's probably overkill.

**Comment by anurse** [comment]
@dotnet-bot test this please

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please
